ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/ScalablePushRegistry.java

```diff
@@ -167,33 +191,40 @@ public synchronized boolean isClosed() {
    * Registers a ProcessingQueue with this scalable push registry so that it starts receiving
    * data as it streams in for that consumer.
    * @param processingQueue The queue to register
-   * @param expectingStartOfRegistryData Whether the request is expecting a new server without
-   *                                     rows received.
+   * @param catchupMetadata The catchup metadata including offsets to start at
    */
   public synchronized void register(
       final ProcessingQueue processingQueue,
-      final boolean expectingStartOfRegistryData
+      final Optional<CatchupMetadata> catchupMetadata
   ) {
     if (closed) {
       throw new IllegalStateException("Shouldn't register after closing");
     }
-    final LatestConsumer latestConsumer = this.latestConsumer.get();
-    if (latestConsumer != null && !latestConsumer.isClosed()) {
-      if (latestConsumer.getNumRowsReceived() > 0
-          && newNodeContinuityEnforced && expectingStartOfRegistryData) {
-        LOG.warn("Request missed data with new node added to the cluster, rows {}",
-            latestConsumer.getNumRowsReceived());
-        throw new KsqlException("New node missed data");
+    try {
+      if (catchupMetadata.isPresent()) {
+        if (catchupCounter.get() >=
+            ksqlConfig.getInt(KsqlConfig.KSQL_QUERY_PUSH_V2_MAX_CATCHUP_CONSUMERS)) {
```

Comment:
general point about this PR - can we at least have JMX metrics around the number of running catchup consumers, the number of active query processing queues attached to the latest queue, etc..? Basically, anything that will help us debug incidents like we've had with hitting the pull query bandwidth limit and it not going down.